spotdl/download/downloader.py

```diff
@@ -24,6 +24,83 @@
 # === Base functionality ===
 # ==========================
 
+# ========================
+# === Helper function ===
+# ========================
+
+def _sanitize_filename(input_str: str) -> str:
+    # ! this is windows specific (disallowed chars)
+    output = input_str
+    for disallowed_char in ['/', '?', '\\', '*', '|', '<', '>']:
+        if disallowed_char in output:
+            output = output.replace(
+                disallowed_char,
+                ''
+            )
+
+    # ! double quotes (") and semi-colons (:) are also disallowed characters but we would
+    # ! like to retain their equivalents, so they aren't removed in the prior loop
+    output = output.replace('"', "'").replace(':', '-')
+
+    return output
+
+
+def _get_smaller_file_path(input_song: SongObj) -> Path:
+    # Only use the first artist if the song path turns out to be too long
+    smaller_name = f"{input_song.get_contributing_artists()[0]} - {input_song.get_song_name()}"
+
+    # this is windows specific (disallowed chars)
+    for disallowed_char in ['/', '?', '\\', '*', '|', '<', '>']:
+        if disallowed_char in smaller_name:
+            smaller_name = smaller_name.replace(disallowed_char, '')
+
+    # ! double quotes (") and semi-colons (:) are also disallowed characters
+    # ! but we would like to retain their equivalents, so they aren't removed
+    # ! in the prior loop
+    smaller_name = _sanitize_filename(smaller_name.replace('"', "'").replace(':', '-'))
```

Comment:
This is more readable

```suggestion
    # ! double quotes (") and semi-colons (:) are also disallowed characters
    # ! but we would like to retain their equivalents, so they aren't removed
    # ! in the prior loop
    smaller_name = smaller_name.replace('"', "'")
    smaller_name = smaller_name.replace(':', '-')

    smaller_name = _sanitize_filename(smaller_name)
```